Bluetooth: Mesh: Generic OnOff & Level Models demo

vikrant8052 opened the pull request:
This is a simple application demonstrating a Bluetooth mesh node in
which Root element has following models

 - Generic OnOff Server
 - Generic OnOff Client
 - Generic Level Server
 - Generic Level Client

And Secondary element has following models

 - Generic OnOff Server
 - Generic OnOff Client

Signed-off-by: Vikrant More vikrant8051@gmail.com

vikrant8052 committed:
Bluetooth: Mesh: Generic OnOff & Level Models demo

This is a simple application demonstrating a Bluetooth mesh node in
which Root element has following models

 - Generic OnOff Server
 - Generic OnOff Client
 - Generic Level Server
 - Generic Level Client

And Secondary element has following models

 - Generic OnOff Server
 - Generic OnOff Client

Signed-off-by: vikrant more vikrant.more@gmail.com

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/sample.yaml:
```diff
@@ -0,0 +1,7 @@
+tests:
+  test:
+    harness: bluetooth
+    platform_whitelist: bbc_microbit nrf51_blenano qemu_x86
```
This sample is specifically for nRF52, so the above whitelist looks wrong to me. It should only contain one or more nRF52 targets. The same applies for the various *.conf files for microbit, etc. There should only be a single prj.conf that targets the nRF52.

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/main.c:
```diff
@@ -0,0 +1,35 @@
+
+	update_light_state();
+
+	printk("Initializing...\n\r");
```
Use only \n for the newlines. You're now using \n in some places and \n\r in others (like here).

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/DEVICE_composition.c:
```diff
@@ -0,0 +1,171 @@
+
+void gen_onoff_get(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx, struct net_buf_simple *buf)
+{
+	doer(model, ctx, buf, 0x8201);
```
You really need to come up with a better name than "doer". `process_message()` perhaps?

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/board.h:
```diff
@@ -0,0 +1,28 @@
+/* board.h - Board-specific hooks */
```
This file seems unnecessary since you're only targetting the nRF52 (i.e. a single board). I know it's there in other samples like samples/bluetooth/mesh, however those are designed to be adapted potentially on multiple different boards.

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/DEVICE_composition.c:
```diff
@@ -0,0 +1,171 @@
+	BT_MESH_MODEL_OP_END,
+};
+
+static const struct bt_mesh_model_op gen_level_srv_op[] = {
```
Move this lower down in the file so that you can make all the handler functions static.

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/microbit.c:
```diff
@@ -0,0 +1,97 @@
+					 { 0, 1, 0, 1, 0 },
+					 { 0, 0, 0, 0, 0 },
+					 { 1, 0, 0, 0, 1 },
+					 { 0, 1, 1, 1, 0 });
```
This image is a smiley face and not an arrow, so better fix the name or the image :) I realize this is probably a copy-paste from somewhere else but let's not replicate the error (instead, please submit a separate fix to the original one).

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/CMakeLists.txt:
```diff
@@ -0,0 +1,22 @@
+	       src/main.c
+	       src/common.c
+	       src/mesh/ble_mesh.c
+               src/mesh/DEVICE_composition.c
```
Why the upper case `DEVICE`? Please be consistent (also, file names are generally lower case in Zephyr). Also, it seems the indentation is off here compared to the previous lines.

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/doer.c:
```diff
@@ -0,0 +1,182 @@
+{
+	uint8_t tid, tmp8;
+	int16_t tmp16;
+	int err = 0, tmp32;
```
If you're going to call it tmp32, better make it s32_t or u32_t.

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/doer.c:
```diff
@@ -0,0 +1,182 @@
+	int err = 0, tmp32;
+
+	struct net_buf_simple *msg;
+
```
Unnecessary empty line

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/doer.c:
```diff
@@ -0,0 +1,182 @@
+	struct server_state *state_ptr = model->user_data;
+
+	switch (opcode) {
+
```
Unnecessary empty line. In fact, it almost looks like you took a different style in this file with lots more empty lines than your other code. Why is that?

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/doer.c:
```diff
@@ -0,0 +1,182 @@
+
+		tmp8 = net_buf_simple_pull_u8(buf);
+
+		state_ptr->current = tmp8;
```
Why is tmp8 needed here? Why not assign to state_ptr->current directly?

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/doer.c:
```diff
@@ -0,0 +1,182 @@
+
+	struct net_buf_simple *msg;
+
+	struct server_state *state_ptr = model->user_data;
```
We don't generally put type information in variable names, so please remove the `_ptr` part.

jhedberg reviewed (changes requested):
There are many more issues I'd like to comment on, but ran out of time for now. I'll take another look once you've sent the next revision.

vikrant8052 committed:
Bluetooth: Mesh: Add persistent storage support in onoff-app

Signed-off-by: vikrant8051 <vikrant8051@gmail.com>

vikrant8052 committed:
Minor updates & coding style improvemnts !!

Signed-off-by: vikrant more <vikrant8051@gmail.com>

vikrant8052 committed:
Minor updates & coding style improvemnts !!

Signed-off-by: vikrant more <vikrant8051@gmail.com>

vikrant8052 committed:
Minor improvements !!

Signed-off-by: vikrant8051 <vikrant8051@gmail.com>

vikrant8052 committed:
Minor improvements !!

Signed-off-by: vikrant8051 <vikrant8051@gmail.com>

vikrant8052 committed:
Minor updates to remove coding style errors !!

Signed-off-by: vikrant more <vikrant8051@gmail.com>

vikrant8052 committed:
Bluetooth: Mesh: Generic OnOff & Level Models demo

Minor updates to remove coding style errors !!

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
Bluetooth: Mesh: Generic OnOff & Level Models demo

Minor updates & coding style improvemnts !!

Signed-off-by: vikrant more <vikrant8051@gmail.com>

vikrant8052 committed:
Bluetooth: Mesh: Generic OnOff & Level Models demo

Minor updates & coding style improvemnts !!

Signed-off-by: vikrant more <vikrant8051@gmail.com>

vikrant8051 commented:
@jhedberg I hope now everything is Okay !! 

There is some error regarding email address. Is that means I can't merge this PR with my personnel email ID ? 
**author email (vikrant8051 <vikrant8051@gmail.com>) does not follow the syntax: First Last <email>.**

How to edti previous commit messages ? 
**failure: commit message error on range: origin/master..HEAD**

jhedberg commented:
> There is some error regarding email address. Is that means I can't merge this PR with my personnel email ID ? author email (vikrant8051 vikrant8051@gmail.com) does not follow the syntax: First Last .

I don't understand what that has to do with your email. The name that precedes the email address is completely independent of the email address. The simplest would be if you just add the following to your `~/.gitconfig`:

```
[user]
	name = Vikrant More
	email = vikrant8051@gmail.com
```

That will then generate the right author information, and the right signed-off-by line when you pass `-s` to git commit.

I also recommend you get familiar with `git rebase -i`

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/process_message.c:
```diff
@@ -0,0 +1,172 @@
+	struct server_state *state = model->user_data;
+
+	switch (opcode) {
+	case 0x8201:	/* GEN_ONOFF_SRV_GET */
```
I took a closer look at this function now (previously named "doer") why are you de-multiplex the message handlers into this single function, when the various case branches have essentially nothing in common (except perhaps some function variables). I'd either split this up to one function per message, or even better just merge the code back to the message handlers that call this function. Why was there a need to do this outside of the message handlers in the first place? (if there is a well justified need I'd still split this up into one function per message, and then you also solve the trouble you had with naming, since you can name the functions based on the message that they handle).

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/publisher.c:
```diff
@@ -0,0 +1,60 @@
+	}
+
+	if (err) {
+		printk("bt_mesh_model_publish: err: %d\n\r", err);
```
still \r here

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/process_message.c:
```diff
@@ -0,0 +1,172 @@
+			err = bt_mesh_model_publish(model);
+
+			if (err) {
+				printk("bt_mesh_model_publish err %d", err);
```
missing \n

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/process_message.c:
```diff
@@ -0,0 +1,172 @@
+			err = bt_mesh_model_publish(model);
+
+			if (err) {
+				printk("bt_mesh_model_publish err %d", err);
```
missing \n

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/ble_mesh.c:
```diff
@@ -0,0 +1,85 @@
+
+	/* Use identity address as device UUID */
+	if (bt_le_oob_get_local(&oob)) {
+		printk("Identity Address unavailable");
```
missing \n

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/mesh/ble_mesh.c:
```diff
@@ -0,0 +1,85 @@
+	struct bt_le_oob oob;
+
+	if (err) {
+		printk("Bluetooth init failed (err %d", err);
```
missing ) and \n

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_app/src/main.c:
```diff
@@ -0,0 +1,38 @@
+		light_state_current.power = LEVEL_100;
+	}
+
+	printk("\n\rled_on_off_status -> %d, led_level_status -> %d\n",
```
still \r here in the beginning. Actually why do you start the format string with a new line? Isn't the one at the end sufficient?

jhedberg commented on samples/boards/nrf52/mesh/onoff-app/prj.conf:
```diff
@@ -2,6 +2,14 @@
+CONFIG_FLASH_MAP=y
+CONFIG_FCB=y
+CONFIG_SETTINGS=y
+CONFIG_SETTINGS_FCB=y
```
Please split this into a separate PR (that's what I intended when I asked for this addition). It can then probably be merged must faster than this PR, since I don't see anything wrong with the patch. If any of this feels like a lot of work, please get familiar with `git rebase -i`. That will get the work done in a matter of seconds, or a minute at most.

vikrant8052 committed:
Bluetooth: Mesh: Generic OnOff & Level Models demo

Removed process_message.c & process_message.h
& merge its content into respective handler functon
in device_composition.c. Plus some minor improvements.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
Bluetooth: Mesh: Generic OnOff & Level Models demo

Removed process_message.c & process_message.h
& merge its content into respective handler functon
in device_composition.c. Plus some minor improvements.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
Bluetooth: Mesh: Generic OnOff & Level Models demo

Removed process_message.c & process_message.h
& merge its content into respective handler functon
in device_composition.c. Plus some minor improvements.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
Bluetooth: Mesh: Generic OnOff & Level Models demo

Removed process_message.c & process_message.h
& merge its content into respective handler functon
in device_composition.c. Plus some minor improvements.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8051 commented:
@jhedberg I make suggested changes. Use ~/.gitconfig to change name. Hope will not face identity error. I will work on Commit message error tomorrow.  Thank You !!

vikrant8052 committed:
Merge branch 'mesh_demo' of https://github.com/vikrant8051/zephyr into mesh_demo

vikrant8052 committed:
Bluetooth: Mesh: Generic OnOff & Level Models demo

Minor updates & coding style improvemnts !!

Signed-off-by: Vikrant More <vikrant8051@gmail.com>
